src/functions/generateRoute.ts

```diff
@@ -1,13 +1,34 @@
 import { useSorobanReact } from '@soroban-react/core';
 import { useFactory } from 'hooks';
 import { useContext, useMemo } from 'react';
-import { CurrencyAmount, Networks, Protocols, Router, Token, TradeType } from 'soroswap-router-sdk';
+import { Currency, CurrencyAmount, Networks, Percent, Protocols, Route, Router, Token, TradeType } from 'soroswap-router-sdk';
 import { AppContext } from 'contexts';
 import { fetchAllPhoenixPairs, fetchAllSoroswapPairs } from 'services/pairs';
+import axios from 'axios';
+import { TokenType } from 'interfaces';
+import { getBestPath, getHorizonBestPath } from 'helpers/horizon/getHorizonPath';
+import { PlatformType } from 'state/routing/types';
+import { CurrencyAmount as AmountAsset } from 'interfaces';
+
+export interface BuildTradeRoute {
```

Comment:
esto deberia extenderse del BuildTradeRoute que viene de soroswap-router-sdk

Ejemplo: 

```ts
// Define the existing interface
interface OldInterface {
  property1: string;
  property2: number;
}

// Extend the existing interface
interface NewExtended extends OldInterface {
  property3: boolean;
  property4?: string; // Optional property
}
```

Reply:
La clase "BuildTradeReturn" de soroswap-router-sdk presenta incompatibilidades de tipo en propiedades que son necesarias para el funcionamiento de la nueva logica:
"Interface 'BuildTradeRoute' incorrectly extends interface 'BuildTradeReturn'.
  Types of property 'amountCurrency' are incompatible.
    Type 'CurrencyAmount | CurrencyAmount<Currency>' is not assignable to type 'CurrencyAmount'.
      Type 'CurrencyAmount' is missing the following properties from type 'CurrencyAmount<Currency>': decimalScale, add, subtract, multiply, and 14 more"